src/jws.rs

```diff
@@ -256,6 +265,13 @@ pub fn verify_bytes(
                     padding = rsa::padding::PaddingScheme::new_pkcs1v15_sign(Some(hash));
                     hashed = crate::hash::sha256(data)?;
                 }
+                Algorithm::PS256 => {
+                    let hash = rsa::hash::Hash::SHA2_256;
+                    let rng = rand_old::rngs::OsRng {};
+                    padding =
+                        rsa::PaddingScheme::new_pss_with_salt::<sha2::Sha256, _>(rng, hash.size());
```

Comment:
I'm a bit confused from what I read online. Don't you need to know the salt/padding for the verification to work instead of recreating it? (I mean, the tests pass, but I don't quite understand why.)

Reply:
Looking at other dependents of the `rsa` crate...
In `jwt-compact`, `PaddingScheme::new_pss_with_salt` is used every time: https://github.com/slowli/jwt-compact/blob/7a99334/src/alg/rsa.rs#L253
In `arweave-rs`, `PaddingScheme::new_pss` is used in verify instead: https://github.com/nestdotland/arweave-rs/blob/79264dc/src/crypto.rs#L72

Reply:
PS256 signature validation (verification) is specified in RFC 7518: https://www.rfc-editor.org/rfc/rfc7518#section-3.5

> The RSASSA-PSS SHA-256 digital signature for a JWS is validated as follows: submit the JWS Signing Input, the JWS Signature, and the public key corresponding to the private key used by the signer to the RSASSA-PSS-VERIFY algorithm using SHA-256 as the hash function and using MGF1 as the mask generation function with SHA-256.

In the alternative implementation using the `ring` crate, verification does not require a random number generator. In the `rsa` crate, it looks like in verification the rng and salt length is discarded:
https://github.com/czzmmc/rsa-rs/blob/ad2417d30b7a94d9d9a49bc86b57c7f0194df7f2/src/key.rs#L206
So it would be equivalent and simpler to use `new_pss` instead of `new_pss_with_salt`; and the rng in this code path does not matter.

To confirm signing is correct... you can generate a PS256 JWT VC now with `example/issue.rs`:
```
cargo run --example issue jwt > vc-ps256.jwt
```
Here is one, verifying with `jwt.io`: [link](https://jwt.io/#debugger-io?token=eyJhbGciOiJSUzI1NiIsImtpZCI6InJzYTIwNDgtMjAyMC0wOC0yNSJ9.eyJpc3MiOiJkaWQ6ZXhhbXBsZTpmb28iLCJ2cCI6eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvMjAxOC9jcmVkZW50aWFscy92MSJdLCJ0eXBlIjoiVmVyaWZpYWJsZVByZXNlbnRhdGlvbiJ9fQ.MqNO2aAfdxP7tFBaEKPHJc4vGEDt_aOZ8o1n5tvcol9NZMozyu7OHltJmEHXRBIVU3LRObm-nOorPshLPrz3etSPDkbzl0zkn8KMq__rq5tYKMQJ__8DXzz7T2F_eGJDLWJUqKHP_JRzZtSSKBb6fCDu-_nKxK-EjuXGSyYe05AteEczRSATjF89ycsGp1i74neZrN0tey7GxPI8d6X_2FVPVvH72OGJ50zmMoV1aHItzOOttTwDT8qdLy6rw9pPZqI1Y6JOU-2PVZMxgNFJZ35l0cPaUSZ2sfQzaP5UL9Ufmf1BZZ-lzFWzLa8fLSK7JbOFWP7arbaqqPJ9qFgLPA&publicKey=-----BEGIN%20PUBLIC%20KEY-----%0AMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAsbX82NTV6IylxCh7MfV4%0AhlyvaniCajuP97GyOqSvTmoEdBOflFvZ06kR%2F9D6ctt45Fk6hskfnag2GG69NALV%0AH2o4RCR6tQiLRpKcMRtDYE%2FthEmfBvDzm%2FVVkOIYfxu%2BIpuo9J%2FS5XDNDjczx2v%2B%0A3oDh5%2BCIHkU46hvFeCvpUS%2BL8TJSbgX0kjVk%2Fm4eIb9wh63rtmD6Uz%2FKBtCo5mmR%0A4TEtcLZKYdqMp3wCjN%2BTlgHiz%2F4oVXWbHUefCEe8rFnX1iQnpDHU49%2FSaXQoud1j%0ACaexFn25n%2BAa8f8bc5Vm%2B5SeRwidHa6ErvEhTvf1dz6GoNPp2iRvm%2BwJ1gxwWJEY%0APQIDAQAB%0A-----END%20PUBLIC%20KEY-----%0A).
Verification can also be done using [jose](https://github.com/latchset/jose): `jose jws ver -i vc-ps256.jwt -k tests/rsa2048-2020-08-25.json`

Reply:
Updated to use `new_pss` instead of `new_pss_with_salt` in verification, for simplicity.